### ISSUE TITLE
Packaging subsystem now works

### ISSUE THREAD

**Issue by deterenkelt** [deterenkelt/Nadeshiko#16]
@RX14 Hi, I’ve implemented proper handling of source subdirectories, now they can be detached safely.

Points of interest
1. New [Makefile](/deterenkelt/Nadeshiko/blob/master/packaging/gentoo/Makefile).
2. Gentoo [ebuild](/deterenkelt/Nadeshiko/blob/master/packaging/gentoo/nadeshiko.ebuild) with all dependencies and USE-flags. The descriptions for USE-flags can be found in [metadata.xml](/deterenkelt/Nadeshiko/blob/master/packaging/gentoo/metadata.xml).
3. The new code for setting the paths to source subdirectories – e.g. `LIBDIR`, `MODULESDIR`, and `EXAMPLECONFDIR` – is here: [bahelite_directories#L97](/deterenkelt/Nadeshiko/blob/v2.5.5/lib/bahelite/bahelite_directories.sh#L97).
**tl;dr** It follows FHS, if the executable is launched from `/usr/{bin,local/bin}`, then the source subdirs are searched in the FHS dirs, if executable is launched from a non-FHS path, then the subdirectories are searched in `MYDIR`. 
4. There was the chicken and egg problem at the very start with sourcing bahelite.sh – it is solved by doing a [simple check](/deterenkelt/Nadeshiko/blob/v2.5.5/nadeshiko.sh#L25) for the path where the executable resides. It’s similar to [the check in bahelite_directories.sh](/deterenkelt/Nadeshiko/blob/v2.5.5/lib/bahelite/bahelite_directories.sh#L276).  

The old Makefile can be found under `packaging/arch/`. If you could find time to update it, that would be great. You can steal the Makefile from the `gentoo` subfolder and modify it for the Arch-friendly make rules. The Arch package can also benefit from updating dependencies according to the ebuild.

**Comment by RX14** [comment]
The gentoo makefile should just be put at the root of the git repo and the arch makefile deleted. The only thing i'd change would be to just install `nadeshiko` and not `nadeshiko.sh` at all.

**Comment by RX14** [comment]
Oh, and the default PREFIX should be `/usr/local` and overrided with `make install DESTDIR=... PREFIX=/usr` in the packaging script.

**Comment by deterenkelt** [comment]
> The only thing i'd change would be to just install nadeshiko and not nadeshiko.sh at all.

I thought about this, and came to a conclusion, that if some end user would like (or need) to dig into the source, it would be better, if their editor could pick syntax highlight without renaming the file to `.sh`. 

Symbolic links in `/usr/bin/` seem to be pretty common, in my current desktop installation there’s 760 of them. So I see no reason why would anybody frown upon Nadeshiko for adding its own.

> Oh, and the default PREFIX should be /usr/local

Hmm, ok I agree. I will try it out to see if it works on Gentoo (99% sure it will, but without a test can’t say), and if it will, then the Makefile would be edited with the default PREFIX set to `/usr/local` and put into the root of the repository. 

The arch folder will be deleted, as it will become empty. If you’d like to commit instructions for the packaging software (akin to Fedora .spec, Debian format/rules or Gentoo .ebuild), feel free to recreate it and make a pull request.

**Comment by deterenkelt** [comment]
Implemented in [2.5.7](/deterenkelt/Nadeshiko/releases/tag/v2.5.7).
